de.dlr.sc.virsat.cef.swtbot.test/src/de/dlr/sc/virsat/cef/swtbot/test/CefHierarchyLevelTest.java

```diff
@@ -0,0 +1,340 @@
+	SWTBotTreeItem elementConfigurationSubSystem;
+	SWTBotTreeItem elementConfigurationEquipment;
+	
+	@Before
+	public void before() throws Exception {
+		super.before();
+		// create the necessary items for the test
+		SwtBotDebugHelper.logCodeLine();
+		repositoryNavigatorItem = bot.tree().expandNode(SWTBOT_TEST_PROJECTNAME, "Repository");
+
+		SwtBotDebugHelper.logCodeLine();
+		configurationTreeSystem = addElement(ConfigurationTree.class, conceptPs, repositoryNavigatorItem);
+		SwtBotDebugHelper.logCodeLine();
+		elementConfigurationSubSystem = addElement(ElementConfiguration.class, conceptPs, configurationTreeSystem);
+		SwtBotDebugHelper.logCodeLine();
+		elementConfigurationEquipment = addElement(ElementConfiguration.class, conceptPs, elementConfigurationSubSystem);
+		SwtBotDebugHelper.logCodeLine();
+
+	}
+	
+	@Test
+	public void testAddSystemParametersViaCheckBox() {
+		
+		openEditor(configurationTreeSystem);
+		SwtBotDebugHelper.logCodeLine();
+		bot.checkBox("System Parameters").click();
+		SwtBotDebugHelper.logCodeLine();
+		waitForEditingDomainAndUiThread();
+		
+		SwtBotDebugHelper.logCodeLine();
+		final int EXPECTED_NUMBER_TREE_CHILDREN_SYSTEM_PARAMETERS = 3; // Documents folder, system parameters, child SEI
+		assertTrue(bot.checkBox("System Parameters").isChecked());
+		SwtBotDebugHelper.logCodeLine();
+		elementConfigurationEquipment.expand();
+		assertThat("There are documents, system parametes and child items now", configurationTreeSystem.getItems(), arrayWithSize(EXPECTED_NUMBER_TREE_CHILDREN_SYSTEM_PARAMETERS));
+		SwtBotDebugHelper.logCodeLine();
+		assertNotEnabled(configurationTreeSystem.contextMenu(conceptCefX.getDisplayName()).menu(getAddCommandName(SystemParameters.class)));
+		
+		SwtBotDebugHelper.logCodeLine();
+		bot.checkBox("System Parameters").click();
+		SwtBotDebugHelper.logCodeLine();
+		waitForEditingDomainAndUiThread();
+		SwtBotDebugHelper.logCodeLine();
+		assertFalse(bot.checkBox("System Parameters").isChecked());
+		SwtBotDebugHelper.logCodeLine();
+		assertThat("System parameters should have been removed", configurationTreeSystem.getItems(), arrayWithSize(2));
+		SwtBotDebugHelper.logCodeLine();
+		assertEnabled(configurationTreeSystem.contextMenu(conceptCefX.getDisplayName()).menu(getAddCommandName(SystemParameters.class)));
+		
+		
+		SwtBotDebugHelper.logCodeLine();
+		bot.checkBox("System Mass Parameters").click();
+		SwtBotDebugHelper.logCodeLine();
+		waitForEditingDomainAndUiThread();
+		SwtBotDebugHelper.logCodeLine();
+		assertTrue(bot.checkBox("System Mass Parameters").isChecked());
+		SwtBotDebugHelper.logCodeLine();
+		final int EXPECTED_NUMBER_TREE_CHILDREN_SYSTEM_MATH_PARAMETERS = 4; // Documents folder, system parameters, system mass parameters, child SEI
+		assertThat("There are documents, system parametes, system mass parameters and child items now", configurationTreeSystem.getItems(), arrayWithSize(EXPECTED_NUMBER_TREE_CHILDREN_SYSTEM_MATH_PARAMETERS));
```

Comment:
Are system parameters added automatically on adding mass parameters? Let's maybe explain it in a comment?